CMakeLists.txt

```diff
@@ -25,6 +25,10 @@ endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules/")
 
+if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
+  add_definitions(-DDEBUG)
```

Comment:
@liupan1111 the same question still holds:

> i don't understand where/how is `DEBUG` macro respected in ceph?

how "-DDEBUG" instructs cmake or make to emits the "-O0"? and you might want to check your cmake installation and inspect ${CMAKE_C_FLAGS_DEBUG} and ${CMAKE_CXX_FLAGS_DEBUG} instead.